File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.swift
import Foundation

func solution(_ common:[Int]) -> Int {
    print(isArithmeticSequence(common))
    if isArithmeticSequence(common) { // 등차수열
        let d = common[common.count-1] - common[common.count-2]
        return common[common.count-1] + d
    } else { // 등비수열
        let r = common[common.count-1] / common[common.count-2]
        return common[common.count-1] * r
    }
}

// 등차수열인지 판단
func isArithmeticSequence(_ arr:[Int]) -> Bool {
    if !arr.count.isMultiple(of: 2) {
        return (arr[0] + arr[arr.count - 1]) == arr[arr.count / 2 ] * 2
    } else {
        return arr[0] + arr[arr.count - 1] == arr[1] + arr[arr.count - 2]
    }
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> common) {
    int answer = 0;

    if((common[0] + common[2]) / 2.0 == common[1]) {
        answer = common.back() + (common[1] - common[0]);
    } else {
        answer = common.back() * (common[1] / common[0]);
    }

    return answer;
}